sanic/app.py

```diff
@@ -876,10 +889,12 @@ async def handle_exception(
         :raises ServerError: response 500
         """
         response = None
-        await self.dispatch(
-            "server.lifecycle.exception",
-            context={"exception": exception},
-        )
+        if not getattr(exception, "__dispatched__", False):
+            ...  # DO NOT REMOVE THIS LINE. IT IS NEEDED FOR TOUCHUP.
```

Comment:
For anyone curious (thanks @ahopkins for the explanation):

> At startup, there are a few performance critical methods that are introspected.
> We generate an AST and walk thru it applying each of a few different operaitons
> One of them looks for a `dispatch` method, then compares the event to the list of registered events in the signal router. If there is no handler, then we remove the `dispatch` from the AST, and then finally recompile the source.
> This way, we do not even call an empty `dispatch` if there will never be a signal on the other end to receive it.
> This has the potential to leave an empty block though. The quick and simple solution is to put an ellipsis or pass in the block. I had been planning to do this dynamically, but never got back to it.
> 
> TL;DR: Ok so the ellipsis is just there in case the dispatch gets removed during touchup because there's no listeners for it.